clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerRecordsTest.java

```diff
@@ -35,28 +35,83 @@ public class ConsumerRecordsTest {
 
     @Test
     public void iterator() throws Exception {
+        String topic = "topic";
+        int recordSize = 10;
+        int partitionSize = 10;
+        int emptyPartitionInterval = 3;
+        ConsumerRecords<Integer, String> records = buildTopicTestRecords(recordSize, partitionSize, emptyPartitionInterval, topic);
+        Iterator<ConsumerRecord<Integer, String>> iterator = records.iterator();
+
+        for (int count = 0; iterator.hasNext(); count++) {
+            if (count % emptyPartitionInterval != 0) {
+                int offset = 0;
+                for (; offset < recordSize; offset++) {
+                    ConsumerRecord<Integer, String> record = iterator.next();
+                    assertEquals(count, record.partition());
+                    assertEquals(topic, record.topic());
+                    assertEquals(offset, record.offset());
+                }
+                assertEquals(recordSize, offset);
+            }
+        }
+    }
 
+    @Test
+    public void testRecordsWithNullTopic() {
+        String nullTopic = null;
         Map<TopicPartition, List<ConsumerRecord<Integer, String>>> records = new LinkedHashMap<>();
+        ConsumerRecords<Integer, String> consumerRecords = new ConsumerRecords<>(records);
```

Comment:
Maybe we can use `ConsumerRecords.empty();`?

Reply:
Cool, I don't know we have this utility function, I will replace with it.